rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistorySpec.scala

```diff
@@ -0,0 +1,123 @@
+package coop.rchain.rspace.nextgenrspace.history
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.test.TestKey32
+import org.scalatest.{FlatSpec, Matchers, OptionValues}
+
+import scala.collection.concurrent.TrieMap
+import History._
+import org.scalacheck.{Arbitrary, Gen, Shrink}
+import coop.rchain.rspace.test.ArbitraryInstances._
+import org.scalatest.prop._
+
+class HistorySpec
+    extends FlatSpec
+    with Matchers
+    with OptionValues
+    with GeneratorDrivenPropertyChecks
+    with InMemoryHistoryTestBase {
+
+  implicit def noShrink[T]: Shrink[T] = Shrink.shrinkAny
+
+  "history" should "accept new leafs (insert, update, delete)" in forAll {
```

Comment:
This test looks quite thorough but it seems to me it doesn't test the internal Trie structure. It would also pass if the Trie was implemented without pointer blocks